QualifiedTeachersApi/src/QualifiedTeachersApi/DataStore/Crm/DataverseAdapter.SetIttResultForTeacher.cs

```diff
@@ -73,6 +73,16 @@ public partial class DataverseAdapter
 
         var (itt, ittLookupFailed) = helper.SelectIttRecord(lookupData.Itt, lookupData.IttProvider.Id);
 
+        if (itt != null && itt.dfeta_Result == dfeta_ITTResult.Withdrawn && result == dfeta_ITTResult.Withdrawn)
```

Comment:
Can these checks go underneath the `if (ittLookupFailed.HasValue)` check? (Seems like they'd be simpler there?)